StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationSummaryBase.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using StyleCop.Analyzers.Helpers;
+
+namespace StyleCop.Analyzers.DocumentationRules
+{
+    public abstract class ElementDocumentationSummaryBase : DiagnosticAnalyzer
+    {
+        abstract protected internal void HandleXmlElement(SyntaxNodeAnalysisContext context, XmlNodeSyntax syntax, params Location[] diagnosticLocations);
```

Comment:
Why is this `protected internal`? It's not called anywhere but this class, so it seems `protected` is the better choince.